server.js

```diff
@@ -1,27 +1,128 @@
 
 // Add middlewares to enable cors and json body parsing
+// access control in the HTTP request
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// Route to get all avocado sales data
+// https://avocado-sales.onrender.com/avocado-sales?page=2
+app.get("/avocado-sales/", (req, res) => {
+  const { page } = req.query;
+  const pageSize = 10; // Set the number of items per page
+
+  // Convert page to a number, default to 1 if not provided or not a valid number
+  const pageNumber = page ? parseInt(page, 10) : 1;
+
+  // Calculate the start and end indices for the slice
+  const startIndex = (pageNumber - 1) * pageSize;
+  const endIndex = pageNumber * pageSize;
+
+  // Slice the data based on the calculated indices
+  const paginatedData = avocadoSalesData.slice(startIndex, endIndex);
+
+  res.status(200).json({
+    success: true,
+    message: "OK ✅",
+    body: { paginatedData },
+  });
+  // res.json(avocadoSalesData);
+});
+
+app.post("/avocado-sales", (req, res) => {
+  // Placeholder for future complex operations to add a new avocado sale
+  res.status(501).json({
+    success: false,
+    message: "Not Implemented ❌",
+    body: { error: "Adding new avocado sale not implemented yet" },
+  });
+});
+// Route to get a list of all registered endpoints
+
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const endpoints = listEndpoints(app);
+  res.json(endpoints);
+});
+
+// Route to get data by ID
+//https://avocado-sales.onrender.com/avocado-sales/2
+app.get("/avocado-sales/:id", (req, res) => {
+  const id = Number(req.params.id);
+  const singleAvocado = avocadoSalesData.find((avocado) => avocado.id === id);
+
+  if (singleAvocado) {
+    res.status(200).json({
+      sucess: true,
+      message: "OK ✅",
+      body: { avocadoSalesData: singleAvocado },
+    });
+  } else {
+    res.status(404).json({
+      sucess: false,
+      message: "Not Found ❌",
+      body: {},
+    });
+  }
+});
+
+// search by region
+//https://avocado-sales.onrender.com/avocado-sales/region/albany
+
+app.get("/avocado-sales/region/:region", (req, res) => {
+  const region = req.params.region;
+
+  if (!region) {
+    return res.status(400).json({
+      success: false,
+      message: "Bad Request: Region parameter is required ❌",
+      body: {},
+    });
+  }
+
+  const filteredRegion = avocadoSalesData.filter(
+    (item) => item.region.toLowerCase() === region.toLowerCase()
+  );
+
+  if (filteredRegion.length === 0) {
+    return res.status(404).json({
+      success: false,
+      message: "Not Found: No data found for the specified region ❌",
+      body: {},
+    });
+  }
+
+  res.status(200).json({
+    success: true,
+    message: "OK ✅",
+    body: { filteredRegion },
+  });
+});
+
+// find the minPrice and maxPrice of average price
+// make a request to https://avocado-sales.onrender.com/avocado-sales/volume/totalVolume?min=0.88&max=0.9
+app.get("/avocado-sales/volume/totalVolume", (req, res) => {
```

Comment:
really nice to get it by min and max price, i'll do the same for this week's assignment! great inspo Miko!